backend/src/services/auth/passportStrategies/githubStrategy.ts

```diff
@@ -0,0 +1,63 @@
+import { get } from 'lodash'
+import GithubStrategy from 'passport-github2'
+import { getServiceChildLogger } from '@crowd/logging'
+import { GITHUB_CONFIG } from '../../../conf'
+import { databaseInit } from '../../../database/databaseConnection'
+import AuthService from '../authService'
+
+const log = getServiceChildLogger('AuthSocial')
+
+export function getGithubStrategy(): GithubStrategy {
+  return new GithubStrategy(
+    {
+      clientID: GITHUB_CONFIG.clientId,
+      clientSecret: GITHUB_CONFIG.clientSecret,
+      callbackURL: GITHUB_CONFIG.callbackUrl,
+      scope: ['user:email'], // Request email scope
+    },
+    (accessToken, refreshToken, profile, done) => {
+      databaseInit()
+        .then((database) => {
+          const email = get(profile, 'emails[0].value')
+          const emailVerified = get(profile, 'emails[0].verified', false)
+          const displayName = get(profile, 'displayName')
+          const { firstName, lastName } = splitFullName(displayName)
+
+          return AuthService.signinFromSocial(
+            'github',
```

Comment:
Since we now have more than one OAuth ways, do u mind putting these string literals into an enumeration and use those values? It would be super cool if u could change the Google strategy to use that enumeration as well, so we don't have actual strings into these functions